src/workers/transactions/userReceivedAward.ts

```diff
@@ -0,0 +1,68 @@
+import { Feature, FeatureType } from '../../entity';
+import {
+  UserTransaction,
+  UserTransactionProcessor,
+} from '../../entity/user/UserTransaction';
+import { NotificationType } from '../../notifications/common';
+import { generateTypedNotificationWorker } from '../notifications/worker';
+
+export const userReceivedAward =
+  generateTypedNotificationWorker<'api.v1.user-transaction'>({
+    subscription: 'api.user-received-award',
+    handler: async (data, con, logger) => {
+      const transaction = await con.getRepository(UserTransaction).findOneBy({
+        id: data.transaction.id,
+      });
+
+      if (!transaction) {
+        logger.error(
+          { transactionId: data.transaction.id },
+          'Transaction not found',
+        );
+        return;
+      }
+
+      if (!transaction.productId) {
+        logger.info(
+          { transactionId: transaction.id },
+          'userReceivedAward: transaction has no productId',
+        );
+        return;
+      }
```

Comment:
No need to log this, because it can happen for Cores purchase, so it would just log unnecessarily.

Reply:
Removed